### PR TITLE
Add documentation about tool dependencies

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR adds information about tool dependencies to the contribution guide.

## Why is it important?

Previously we did not specify what tools were required to run `go generate` in the repo.

## Checklist

- [x] I have made corresponding changes to the documentation
~~- [ ] My code follows the style guidelines of this project~~
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

## Related issues

https://github.com/elastic/beats/pull/15998

### CODE REVIEW

File: docs/devguide/contributing.asciidoc [hunk 1]
```diff
@@ -172,12 +172,30 @@ your browser with the docs for preview.
...
 
+==== Non-Golang dependencies
+
+Besides Go libraries, we are using development tools to generate parsers for inputs and processors.
+
+The following packages are required to run `go generate` by Beat:
+
+===== Auditbeat
+
+* Flatbuffers >= 1.9
```
Comment (andrewkroh) [inline docs/devguide/contributing.asciidoc]:
```suggestion
* FlatBuffers >= 1.9
```

File: docs/devguide/contributing.asciidoc [hunk 2]
```diff
@@ -172,12 +172,30 @@ your browser with the docs for preview.
...
 
+==== Golang
+
 To manage the `vendor/` folder we use
 https://github.com/kardianos/govendor[govendor]. Please see
 the govendor documentation on how to add or update vendored dependencies.
 
 In most cases `govendor fetch your/dependency@version +out` will get the job done.
 
+==== Non-Golang dependencies
```
Comment (dedemorton) [inline docs/devguide/contributing.asciidoc]: I'd just say "Other dependencies"

File: docs/devguide/contributing.asciidoc [hunk 3]
```diff
@@ -172,12 +172,30 @@ your browser with the docs for preview.
...
+
+Besides Go libraries, we are using development tools to generate parsers for inputs and processors.
+
+The following packages are required to run `go generate` by Beat:
```
Comment (dedemorton) [inline docs/devguide/contributing.asciidoc]: Not sure what you mean when you say "by Beat"? Do you mean when you run `go generate` under for a specific Beat?

File: docs/devguide/contributing.asciidoc [hunk 4]
```diff
@@ -172,12 +172,30 @@ your browser with the docs for preview.
 [[dependencies]]
 === Dependencies
```
Comment (dedemorton) [inline docs/devguide/contributing.asciidoc]: Try adding a sentence intro here. I think that will fix your doc build error.

**Comment by kvch** [comment]
Docs build is passing, merging.

**Pull request merged**